src/onegov/search/indexer.py

```diff
@@ -392,17 +394,13 @@ def bulk_process(self):
         Gather all index tasks, group them by model and index batch-wise
         """
 
-        while not self.queue.empty():
-            tasks = {}
-
-            for _ in range(self.queue.qsize()):
-                task = self.queue.get(block=False, timeout=None)
-                if task['action'] == 'index':
-                    tasks.setdefault(task['tablename'], []).append(task)
-                self.queue.task_done()
+        def task_generator():
+            while not self.queue.empty():
+                yield self.queue.get(block=False, timeout=None)
```

Comment:
You accidentally got rid of `self.queue.task_done()`.

Reply:
Absolutely!

Reply:
I'm not sure if you're allowed to call `task_done` before pulling it off the queue, so I'd reorder these two statements, if you're worried about the final task of a group hanging around while it's being processed you can pull the task out into a local variable before yielding it:
```python
                task = self.queue.get(block=False, timeout=None)
                self.queue.task_done()
                yield task
```